sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java

```diff
@@ -182,10 +209,21 @@ protected void newLine() {
     @Override
     protected void eof() {
       if (this.line > 0) {
-        consumer.consume(line, sb.length() > 0 ? lineMd5Digest.digest(sb.toString().getBytes(StandardCharsets.UTF_8)) : null);
+        processBuffer();
       }
     }
 
+    private void processBuffer() {
+      try {
+        if (sb.length() > 0) {
+          ByteBuffer encoded = encoder.encode(CharBuffer.wrap(sb));
+          lineMd5Digest.update(encoded.array(), 0, encoded.limit());
+          consumer.consume(line, lineMd5Digest.digest());
+        }
+      } catch (CharacterCodingException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)